specifyweb/frontend/js_src/lib/components/DataModel/legacyTypes.ts

```diff
@@ -43,20 +58,22 @@ export type SpecifyResource<SCHEMA extends AnySchema> = {
    */
   /* eslint-disable @typescript-eslint/method-signature-style */
   get<
-    FIELD_NAME extends
-      | keyof CommonFields
-      | keyof SCHEMA['fields']
-      | keyof SCHEMA['toManyDependent']
-      | keyof SCHEMA['toManyIndependent']
-      | keyof SCHEMA['toOneDependent']
-      | keyof SCHEMA['toOneIndependent'],
-    VALUE extends (CommonFields &
-      IR<never> &
-      SCHEMA['fields'] &
-      SCHEMA['toManyDependent'] &
-      SCHEMA['toManyIndependent'] &
-      SCHEMA['toOneDependent'] &
-      SCHEMA['toOneIndependent'])[FIELD_NAME]
+    // FIELD_NAME extends
+    //   | keyof CommonFields
+    //   | keyof SCHEMA['fields']
+    //   | keyof SCHEMA['toManyDependent']
+    //   | keyof SCHEMA['toManyIndependent']
+    //   | keyof SCHEMA['toOneDependent']
+    //   | keyof SCHEMA['toOneIndependent'],
+    // VALUE extends (CommonFields &
+    //   IR<never> &
+    //   SCHEMA['fields'] &
+    //   SCHEMA['toManyDependent'] &
+    //   SCHEMA['toManyIndependent'] &
+    //   SCHEMA['toOneDependent'] &
+    //   SCHEMA['toOneIndependent'])[FIELD_NAME]
+    FIELD_NAME extends AllowedFields<SCHEMA>,
```

Comment:
Instead, here is what you could do:
Instead of creating a new type, AllowedField, use `keyof CommonFields | TableFields<SCHEMA>`
(TableFields is a type defined in `components/DataModel/helperTypes.ts`

Then, don't create a new type for AllowedValues, but leave it inlined as it was - since it's just 8 lines, the jscpd should not trip up anymore. If it does, then add a comment to silence it.

The reason I am against creating `AllowedValues` is that it would be more complicated to de-duplicate the code than it's worth (i.e, you would need to pass both SCHEMA and FIELD_NAME to AllowedValues - which might work, but TypeScript might not be able to handle that well as this

Reply:
Actually, I am still getting typescript errors even with that change:

```ts
FIELD_NAME extends keyof CommonFields | keyof TableFields<SCHEMA>,
```

let's just leave this file as is and add a comment to make jscdp ignore the copy/paste. This file is called legacyTypes.ts because this will be deleted once we transition away from the legacy librarty